pkg/util/validation/limits.go

```diff
@@ -1151,6 +1153,43 @@ func MaxDurationPerTenant(tenantIDs []string, f func(string) time.Duration) time
 	return result
 }
 
+// PreferredStringPerTenant returns the preferred string value among the actual settings for the
+// given tenants. The order of preference is specified in the input preferences slice. A preference
+// with lower index means higher importance. If no setting matches the any of the specified preferences,
+// then the value from the first tenant is returned.
+func PreferredStringPerTenant(tenantIDs []string, f func(string) string, preferences []string) string {
```

Comment:
the godoc doesn't mention the function parameter. Can you include that too?

Reply:
and also - i don't quite get the point of this function. Can't we just say "if any tenant has strong read consistency, then the query uses that, otherwise use weak consistency"?

Reply:
> Can't we just say "if any tenant has strong read consistency, then the query uses that, otherwise use weak consistency"?

Yes, but that's what the function does. The function is just a generalization.

Reply:
> the godoc doesn't mention the function parameter. Can you include that too?
Done in https://github.com/grafana/mimir/pull/8808/commits/b7eebcda87d416801555f8dabe05ce0c60543b34

Reply:
it looks overcomplicated. Is there another place which needs this kind of logic? Wouldn't something like this be simpler

```go
level = querierapi.ReadConsistencyEventual
for _, t := range tenantIDs {
  if r.limits.IngestStorageReadConsistency(t) == querierapi.ReadConsistencyStrong {
    consistency = querierapi.ReadConsistencyStrong
    break
  }
}
```

Reply:
Ok. Simplified as you suggested here: https://github.com/grafana/mimir/pull/8808/commits/bb45335c6bf64289e24e13f5f2ad4344727dcb78